File: FizzBuzz-ruby/lib/fizzbuzz.rb
class Fizzbuzz

	def calc ( number )
		num = number
		output = num
		output = 'Fizz' if num % 3 == 0
		output = 'Buzz' if num % 5 == 0
		output = 'FizzBuzz' if num % 15 == 0

		output
	end
end

File: fizzbuzz-javascript/src/fizzbuzz.js
function Fizzbuzz(){

  Fizzbuzz.prototype.isDivisibleByThree = function(number){
    return number % 3 === 0;
  };

  Fizzbuzz.prototype.isDivisibleByFive = function(number){
    return number % 5 === 0;
  };

  Fizzbuzz.prototype.isDivisibleByFifteen = function(number){
    return number % 15 === 0;
  };

  Fizzbuzz.prototype.convert = function(number){
    if (this.isDivisibleByFifteen(number)) return "FizzBuzz";
    if (this.isDivisibleByThree(number)) return "Fizz";
    if (this.isDivisibleByFive(number)) return "Buzz";
    return number;
  };
}
